db/schema.sql

```diff
@@ -170,7 +170,8 @@ CREATE TABLE collect_jobs(
     id                      BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,  -- artificial ID, internal-only
     collect_job_id          UUID NOT NULL,               -- UUID used by collector to refer to this job
     task_id                 BIGINT NOT NULL,             -- the task ID being collected
-    batch_interval          TSRANGE NOT NULL,            -- the batch interval, as a range of timestamps
+    batch_identifier        BYTEA NOT NULL,              -- encoded query-type-specific batch identifier (corresponds to identifier in BatchSelector)
+    batch_interval          TSRANGE,                     -- batch interval, as a TSRANGE, populated only for time-interval tasks. (will always match batch_identifier)
```

Comment:
Column `batch_identifier` now occurs in table `collect_jobs` and `aggregate_jobs`, and I would bet you'll need it in `aggregate_share_jobs` when you do the helper part of this work. Should we introduce a `batches` table so that `aggregate_jobs` and `collect_jobs` or `aggregate_share_jobs` (depending on aggregator role) can consistently refer to the same batches?

Reply:
Sure -- there's no technical requirement to do so, but I suppose not writing down the batch identifiers over & over again might be nice (we certainly arrange so that other DAP identifiers are typically only written down in one row).

This also allows me to move the batch interval logic to a central location.

Reply:
SGTM, we'll have a better perspective on all this once the helper aggregate share stuff is in place. Go ahead and merge at your convenience.